public/app/features/datasources/state/contexts.ts

```diff
@@ -0,0 +1,8 @@
+import { createContext } from 'react';
+
+import { DATASOURCES_ROUTES } from '../constants';
+import { DataSourcesRoutes } from '../types';
+
+// The purpose of this context is to be able to override the data-sources routes (used for links for example) used under
```

Comment:
Thank you for writing this. As I was scanning other files in this PR I was questioning the usage of this context. 👏